File: python/Patching Array.py
def minPatches(nums, n):
    patch = 0
    count = 0
    index = 0
    while patch < n:
        if index < len(nums) and patch + 1 >= nums[index]:
            patch += nums[index]
            index += 1
        else:
            patch += (patch + 1)
            count += 1
    return count
print(minPatches([1,3],  6))

File: JAVA/java/src/PatchingArray.java
public class PatchingArray {
    public static  int minPatches(int[] nums, int n) {
        int count = 0;
        long missing = 1;
        int i = 0;

        while (missing <= n) {
            if (i < nums.length && nums[i] <= missing) {
                missing += nums[i++];
            } else {
                missing += missing;
                count++;
            }
        }

        return count;
    }

    public static void main(String[] args) {
        System.out.println(minPatches(new int[]{1,3}, 6));
    }
}
